enhancements/microshift/multus-cni-for-microshift.md

```diff
@@ -0,0 +1,445 @@
+                        "dns": {},
+                        "gateway": [
+                            "\u003cnil\u003e"
+                        ]
+                    },{
+                        "name": "default/bridge-conf",
+                        "interface": "net2",
+                        "ips": [
+                            "192.168.20.203"
+                        ],
+                        "mac": "7a:ca:cf:19:64:e8",
+                        "dns": {},
+                        "gateway": [
+                            "\u003cnil\u003e"
+                        ]
+                    }]
+                  k8s.v1.cni.cncf.io/networks: bridge-conf,bridge-conf
+```
+
+There is an annotation for NetworkAttachmentDefinition - `k8s.v1.cni.cncf.io/resourceName`.
+It is used when some CNI requires information about specific device that is prepared by device plugin.
+It is primarily used with SR-IOV and as such out of scope for this enhancement mentioned only for completeness.
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+First, it must be noted that Multus CNI itself is a meta-CNI. From a high level perspective, its
+purpose is to call other CNIs according to the CNI configs supplied by the user in form of
+NetworkAttachmentDefinitions and Pod annotations.
+Any specific actions are related to the delegate CNIs. For example: creating veth pair,
+attaching one end of veth to the bridge and making the other end available within the Pod is
+responsibility of `bridge` CNI is this example.
+MicroShift team will create tests for CNIs that will be declared as supported. However these tests
+will not explore the breadth and depth of possible network setups, so ultimately the responsibility
+for correctness of the configuration is up to the user.
+
+#### Manifests
+
+MicroShift will provide Multus based on thin architecture because of the resource consumption
+(see alternatives for more information).
+Because of the differences with OpenShift (which uses thick Multus plugin), existing OpenShift
+manifests will require changes to make them suitable for MicroShift. These manifests will reside
+in MicroShift repository.
+
+Updating necessary image references will be part of existing rebase procedure.
+
+#### RPM package
+
+RPM spec to build `microshift-multus` and `microshift-multus-release-info` RPMs will be part of
+existing `microshift.spec` file.
+The RPM will include:
+- manifests required to deploy Multus on MicroShift
+- CRI-O drop-in config to use Multus instead of ovn-kubernetes (which will require reorganization
+  of currently existing MicroShift's CRI-O configs)
+- greenboot healthcheck script
+- cleanup script plugin
+
+#### Container images: Multus and network plugins
+
+Because the Multus image used by OpenShift has a large size that is not acceptable for edge devices, new image
+will be prepared and will only include relevant artifacts such as entrypoint script/binary and
```

Comment:
Will these new images be part of openshift payload? If so, it would be good to call it out.

Reply:
done